drafts/element-internals-api.md

```diff
@@ -0,0 +1,179 @@
+---
+title: Finally, custom form elements that don't suck!
+domain: campzulu.hashnode.dev
+tags: web-development, lit, web-components, shadow-dom, form
+cover: https://cdn.hashnode.com/res/hashnode/image/stock/unsplash/jJT2r2n7lYA/upload/76adc03462239d3b3e6cb53f44b60b20.jpeg?w=1600&h=840&fit=crop&crop=entropy&auto=compress,format&format=webp
+publishAs: r3dDoX
+hideFromHashnodeCommunity: false
+ignorePost: true
+---
+
+In the world of web development, creating custom form elements has always been a bit of a challenge.
+However, with the introduction of web components and the use of the ElementInternals API, this task
```

Comment:
What I found confusing about ElementInternals, is that they are specifically made for form elements, although the name does not say anything about forms. Maybe we could add a note to that (when first reading the post, I thought ElementInternals could be used for other things too)

Reply:
I called it "attach" now. This way we use the same naming. There is not much more details to go into without going into Browser internals. So I hope this makes it clear enough